pytorch_lightning/loggers/mlflow.py

```diff
@@ -151,6 +151,11 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
             if isinstance(v, str):
                 log.warning(f'Discarding metric with string value {k}={v}.')
                 continue
+            elif '(' in k or ')' in k:
+                # MLFlow does not support parenthesis in metric name.
+                sk = k.replace('(', '').replace(')', '')
+                log.warning(f'MLFlowLogger does not support parenthesis in metric name.\nReplacing {k} with {sk}.')
```

Comment:
will it appear again and again everytime it tried to log something with these keys. If it will then I don't think a warning is good here.

Reply:
warnings.warn() will print only once
log.warning will print everytime

@dscarmo if you have the time, could we aim for an implementation of an sanitize function that takes care of all invalid symbols?
Here is my quick incomplete attempt from the top of my head:
```python 
import re
k = re.sub("[^a-zA-Z0-9]", "", k)
```

Reply:
Updated the PR followin @awaelchli's suggestion, adding supported characters.
The regex ("[^a-zA-Z0-9_/. -]+") can be tested in https://regexr.com/.

Apologize if im doing something wrong and thats why tests are failing. I'm new to contributing to public repos.

Reply:
@dscarmo failing tests are unrelated to this PR.